File: android/app/src/main/kotlin/com/groupany/mangatek_flutter/pigeon/FlavorApiImpl.kt
package com.groupany.mangatek_flutter.pigeon

import android.content.Context
import com.groupany.mangatek_flutter.BuildConfig
import com.groupany.mangatek_flutter.R

class FlavorApiImpl(private val context: Context) : FlavorApi {
    override fun getFlavor(): String = BuildConfig.FLAVOR

    override fun getAppName(): String = context.getString(R.string.app_name)

    override fun isProd(): Boolean = getFlavor() == "prod"
}

File: ios/Runner/Pigeons/FlavorApiImpl.swift
import Foundation

class FlavorApiImpl: FlavorApi {
    func getFlavor() throws -> String {
        return Bundle.main.infoDictionary?["APP_FLAVOR"] as? String ?? ""
    }

    func getAppName() throws -> String {
        return Bundle.main.infoDictionary?["CFBundleDisplayName"] as? String ?? ""
    }

    func isProd() throws -> Bool {
        return (try getFlavor()) == "prod"
    }
}
